mso/resource_mso_schema_site_anp_epg_bulk_staticport.go

```diff
@@ -0,0 +1,749 @@
+			},
+			"site_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"template_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"anp_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"epg_name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"static_ports": &schema.Schema{
+				Type: schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "port",
+							ValidateFunc: validation.StringInSlice([]string{
+								"port",
+								"vpc",
+								"dpc",
+							}, false),
+						},
+						"pod": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"leaf": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"path": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ForceNew:     true,
+							ValidateFunc: validation.StringLenBetween(1, 1000),
+						},
+						"vlan": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"deployment_immediacy": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"immediate",
+								"lazy",
+							}, false),
+						},
+						"fex": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+						"micro_seg_vlan": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Computed: true,
+						},
+						"mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"native",
+								"regular",
+								"untagged",
+							}, false),
+						},
+					},
+				},
+				Required: true,
+			},
+		}),
+	}
+}
+
+func resourceMSOSchemaSiteAnpEpgBulkStaticPortImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+
+	msoClient := m.(*client.Client)
+
+	get_attribute := strings.Split(d.Id(), "/")
+	schemaId := get_attribute[0]
+	stateSite := get_attribute[2]
```

Comment:
I see it is done consistently (not saying we shouldn't do it)  but just out of curiosity why do we prepend with state?

Reply:
resolved